pkg/syncer/cnsoperator/manager/init.go

```diff
@@ -69,11 +69,15 @@ func InitCnsOperator(ctx context.Context, clusterFlavor cnstypes.CnsClusterFlavo
 	log.Infof("Initializing CNS Operator")
 	cnsOperator := &cnsOperator{}
 	cnsOperator.configInfo = configInfo
-	vCenter, err := types.GetVirtualCenterInstance(ctx, cnsOperator.configInfo, false)
-	if err != nil {
-		return err
+
+	var volumeManager volumes.Manager
+	if clusterFlavor == cnstypes.CnsClusterFlavorWorkload || clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
+		vCenter, err := types.GetVirtualCenterInstance(ctx, cnsOperator.configInfo, false)
+		if err != nil {
+			return err
+		}
+		volumeManager = volumes.GetManager(ctx, vCenter)
```

Comment:
I later in the code we are using volumeManager
Can you make sure to add a check for non nil, to avoid NPE

```
	// Setup all Controllers
	if err := controller.AddToManager(mgr, clusterFlavor, cnsOperator.configInfo, volumeManager); err != nil {
		log.Errorf("failed to setup the controller for Cns operator. Err: %+v", err)
		return err
	}
```

Reply:
We will not get NPE expection at all as none of the controller are used for GC.

Reply:
since this is not initialized for some cases, I suggest adding checks before using volumeManager.